src/Tests/SmsFrameworkDeliveryReportTest.php

```diff
@@ -75,49 +68,88 @@ public function testDeliveryReports() {
   public function testDeliveryReportUpdate() {
```

Comment:
Moved the test to a kernel test and also tested processing a delivery report (which is really what we are testing here).
The automatic delivery report route will be done in another issue (I think there is an existing one already).